distribution/pull_v2.go

```diff
@@ -208,7 +208,9 @@ func (ld *v2LayerDescriptor) Read(p []byte) (int, error) {
 }
 
 func (ld *v2LayerDescriptor) Close() {
-	ld.layerDownload.Close()
+	if ld.layerDownload != nil {
```

Comment:
Is there a static analysis tool to find such dereferencing (`pointer.dataMameber.\w+`)